ngraph/src/ngraph/frontend/onnx_import/op/upsample.cpp

```diff
@@ -0,0 +1,183 @@
+// Copyright 2017-2020 Intel Corporation
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//*****************************************************************************
+
+#include <memory>
+
+#include "default_opset.hpp"
+#include "exceptions.hpp"
+#include "upsample.hpp"
+
+namespace ngraph
+{
+    namespace onnx_import
+    {
+        namespace op
+        {
+            namespace
+            {
+                bool check_mode_support(const onnx_import::Node& node, const std::string mode)
+                {
+                    std::unordered_set<std::string> supported_modes = {"nearest", "linear"};
+                    bool is_mode_supported =
+                        (std::find(supported_modes.begin(), supported_modes.end(), mode) !=
+                         supported_modes.end());
+
+                    if (!is_mode_supported)
+                    {
+                        std::string supported_modes_str = "";
+                        for (const auto& mode_name : supported_modes)
+                        {
+                            supported_modes_str += (mode_name + ", ");
+                        }
+                        CHECK_VALID_NODE(node,
+                                         is_mode_supported,
+                                         mode,
+                                         " - this type of interpolation mode is not supported."
+                                         " Choose one of the following modes: ",
+                                         supported_modes_str);
+                    }
+                    return is_mode_supported;
+                }
+            }
+
+            namespace set_1
+            {
+                NodeVector upsample(const onnx_import::Node& node)
+                {
+                    const auto inputs = node.get_ng_inputs();
+                    const auto data = inputs.at(0);
+
+                    const auto data_shape = data->get_output_partial_shape(0);
+
+                    const auto scales = node.get_attribute_value<std::vector<float>>("scales");
+                    const auto mode = node.get_attribute_value<std::string>("mode", "nearest");
+                    check_mode_support(node, mode);
+
+                    size_t axes_size = scales.size();
+
+                    AxisSet axes;
+                    for (int ax = 0; ax < axes_size; ++ax)
+                    {
+                        axes.insert(ax);
+                    }
```

Comment:
Consider using `std::iota` here...
```suggestion
    AxisSet axes(axes_size);
    std::iota(axes.begin(), axes.end(), 0);
```

Reply:
Sure, I thought about it, but I wasn't sure is that makes sense for `unordered_set`.

Reply:
Actually, to use `std::iota` in this case we would need `std::vector` help. 
`AxisSet` inherits from `std::set` so it doesn't even have constructor that take size like you proposed or reserve function.